File: Easy/1200_Minimum_Absolute_Difference/Solution.py
from typing import List, Optional


class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr.sort()
        min_diff = float('inf')
        result = []

        for i in range(1, len(arr)):
            diff = arr[i] - arr[i - 1]
            if diff < min_diff:
                min_diff = diff
                result = [[arr[i - 1], arr[i]]]
            elif diff == min_diff:
                result.append([arr[i - 1], arr[i]])

        return result


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    print(solution.minimumAbsDifference([4, 2, 1, 3]))

    # test cases 2
    print(solution.minimumAbsDifference([1, 3, 6, 10, 15]))

    # test cases 3
    print(solution.minimumAbsDifference([3, 8, -10, 23, 19, -4, -14, 27]))

File: Easy/1200_Minimum_Absolute_Difference/Solution.java
import java.util.List;
import java.util.Arrays;
import java.util.ArrayList;


public class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        Arrays.sort(arr);
        int minDiff = Integer.MAX_VALUE;
        List<List<Integer>> result = new ArrayList<>();

        for (int i = 1; i < arr.length; i++) {
            int diff = arr[i] - arr[i - 1];
            if (diff < minDiff) {
                minDiff = diff;
                result.clear();
                result.add(Arrays.asList(arr[i - 1], arr[i]));
            } else if (diff == minDiff) {
                result.add(Arrays.asList(arr[i - 1], arr[i]));
            }
        }

        return result;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        List<List<Integer>> result1 = solution.minimumAbsDifference(new int[]{4, 2, 1, 3});
        System.out.println(result1);

        // test cases 2
        List<List<Integer>> result2 = solution.minimumAbsDifference(new int[]{1, 3, 6, 10, 15});
        System.out.println(result2);

        // test cases 3
        List<List<Integer>> result3 = solution.minimumAbsDifference(new int[]{3, 8, -10, 23, 19, -4, -14, 27});
        System.out.println(result3);

    }
}